lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -1820,14 +1820,16 @@ class SqlParser(private val ion: IonSystem) : Parser {
         return ParseNode(ParseType.CALL, name, arguments, rem.tail)
     }
 
-    private fun List<Token>.expectDatePart(): ParseNode =
-        when (this.head?.type) {
-            DATE_PART -> {
-                val datePartToken = this.head!!
-                ParseNode(ATOM, datePartToken, listOf(), this.drop(1))
+    private fun List<Token>.parseDatePart(): ParseNode {
+        val maybeDatePart = this.head
+        return when  {
+            maybeDatePart?.type == IDENTIFIER && DATE_PART_KEYWORDS.contains(maybeDatePart.text?.toLowerCase()) -> {
+                val datePartToken = maybeDatePart!!
```

Comment:
(Taken from Intellij) Now that `maybeDatePart` is checked for being null on L1826, can remove the `!!` and can inline `maybeDartPart` directly into the `ParseNode`.

```kotlin
ParseNode(ATOM, maybeDatePart, listOf(), this.tail)
```

Reply:
Thx. done.